AK/String.h

```diff
@@ -27,8 +27,8 @@ namespace Detail {
 class StringData;
 }
 
-// FIXME: Remove this when Apple Clang fully supports consteval.
-#if defined(AK_OS_MACOS)
+// FIXME: Remove this when Apple Clang and OpenBSD Clang fully supports consteval.
+#if defined(AK_OS_MACOS) || defined(AK_OS_OPENBSD)
```

Comment:
Hmm. Is there a feature test macro we can use here instead of just assuming specific compiler versions for specific OS's? <https://en.cppreference.com/w/User:D41D8CD98F/feature_testing_macros>

Reply:
There's `__cpp_consteval`, but clang-15 doesn't define it yet, even though clang-15 is happy with this function. It's still listed as "partial" support on <https://en.cppreference.com/w/cpp/compiler_support>.

Reply:
Fair enough. hopefully it gets properly added in the clang 17 timeframe and we can rely on that define instead of OS checks at that time.